crates/bitwarden/src/client/client.rs

```diff
@@ -69,20 +69,24 @@ pub(crate) enum ServiceAccountLoginMethod {
 /// The main struct to interact with the Bitwarden SDK.
 #[derive(Debug)]
 pub struct Client {
-    token: Option<String>,
-    pub(crate) refresh_token: Option<String>,
-    pub(crate) token_expires_on: Option<i64>,
-    pub(crate) login_method: Option<LoginMethod>,
+    token: RwLock<Option<String>>,
+    pub(crate) refresh_token: RwLock<Option<String>>,
+    pub(crate) token_expires_on: RwLock<Option<i64>>,
```

Comment:
I wonder if we should find some way to group these together. You shouldn't be able to write to just one of these.

Reply:
Yeah, that's true, I extracted them to a tokens struct, which at least simplifies a bit all the locking and unlocking.